Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -162,7 +162,17 @@ template <class Type> class TableColumn : public API::Column {
         typename boost::mpl::if_c<boost::is_convertible<double, Type>::value,
                                   Type, InconvertibleToDoubleType>::type
             DoubleType;
-    return boost::numeric_cast<double, DoubleType>(m_data[i]);
+    if (typeid(m_data[i]) == typeid(std::string)) {
```

Comment:
To be honest I don't really like the reliance on `typeid` at all. It's a known problem to use them across DLL boundaries and the code ends up with lots of `if` statements like this. It would be better to use some type of `traits` approach but I understand that overhauling that is outside of the scope of a fix for this.

I would like us to remove the usage of `boost::mpl` and switch to using the `std` `type_traits`.